src/clts2vec/parse.py

```diff
@@ -53,29 +58,31 @@ class CLTS2Vec:
     ts = None
 
     def __init__(
-            self, 
-            ts=None, 
+            self,
+            ts=None,
             feature_values=None,
             feature_hierarchy=None,
             binary_features=None,
             joint_defs=None
-            ):
-        
+    ):
+
         self.ts = ts
         self.feature_values = feature_values or self.feature_values
         self.joint_defs = joint_defs or self.joint_defs
         self.feature_hierarchy = feature_hierarchy or self.feature_hierarchy
         self.binary_features = binary_features or self.binary_features
 
     def __call__(self, sounds, vectorize=True):
-        return [self.parse(sound, vectorize=vectorize) for sound in sounds]
+        return [self.get_vec(sound, vectorize=vectorize) for sound in sounds]
 
     def validate(self, sound):
         """
         Try to retrieve the sound name from CLTS or `linse`.
         """
-        if is_valid_sound(sound):
-            return sound
+        if not sound:
```

Comment:
I prefer the validator above here. The advantage of the little function above is that it can also be used in other contexts. So doing all the type checks now here is not needed, if you modify the typecheck above. So is_valid_sound should return False in case of passing it a non-string.

Reply:
sure, that makes sense. moved the type-check to `is_valid_sound`.

Reply:
Nevermind, I had a good reason for putting the check there: The method breaks when passing `None` as `sound`, since the transcription system cannot handle it. When removing the check, the code fails with the following exception (generated inside CLTS):

```
AttributeError: 'NoneType' object has no attribute 'split'
```

I think it would be better to account for that case by explicitly checking for it and throwing an exception accordingly.

Reply:
That's why the `is_valid_sound`-check should check also that `sound` is string, as I mentioned above ;-)

Reply:
If you are fine, I just take over and fix this then, with tests in place, this is easy to verify ;-)

Reply:
`is_valid_sound` is not the problem, but `validate`:

```python
    def validate(self, sound):
        """
        Try to retrieve the sound name from CLTS or `linse`.
        """
        if hasattr(sound, "name") and is_valid_sound(sound.name):
            return sound.name
        if self.ts:
            sound = self.ts([sound])[0]  # here the system tries to parse 'sound' (which breaks if 'sound' is None)
            if hasattr(sound, "name"):
                sound = sound.name
            if is_valid_sound(sound):
                return sound
        raise ValueError("Invalid sound encountered.")
```

Reply:
I'll push my most recent changes, then you can look into it.